#!/usr/bin/env python

PACKAGE = "vsemi_c320_camera"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

lens_type_enum = gen.enum([ gen.const("WideField",      int_t, 0, "WideField"),
                            gen.const("StandardField",  int_t, 1, "StandardField"),
                            gen.const("NarrowField",    int_t, 2, "NarrowField")
                            ],
                            "An enum to set lens type")


image_type_enum = gen.enum([ gen.const("Grayscale", int_t, 0, "Grayscale"),
                             gen.const("Distance",    int_t, 1, "Distance"),
                             gen.const("DistranceAmplitude", int_t, 2, "Distance&Amplitude"),
                             gen.const("DCS", int_t, 3, "DCS")
                             ],
                             "An enum to set image type")
                             
frequence_modulation_enum = gen.enum([ gen.const("24MHz", int_t, 0, "24MHz"),
                                       gen.const("12MHz", int_t, 1, "12MHz"),
                                       gen.const("6MHz", int_t, 2, "6MHz"),
                                       gen.const("3MHz", int_t, 3, "3MHz"),
                                       gen.const("1_5MHz", int_t, 4, "1_5MHz"),
                                       gen.const("0_75MHz", int_t, 5, "0_75MHz")],
                                       "An enum to set modulation frequency")

######  Name                    Type   Reconfiguration level  Description     Default Min  Max
### Acquisition Control
gen.add("lens_type",              int_t,  0,  "Lens type",  0, 0, 2, edit_method=lens_type_enum)
gen.add("image_type",             int_t,  0,  "Image type", 1, 0, 2, edit_method=image_type_enum)
gen.add("frequency_modulation",   int_t,  0,  "Frequency modulation",                  2,  2, 3, edit_method=frequence_modulation_enum)
gen.add("integration_time_tof_1", int_t,  0,  "Integration time TOF1 [uS]",         1200,  0, 4000)
gen.add("min_amplitude",          int_t,  0,  "threshold minAmplitude LSB",           60,  0, 100000)
#gen.add("median_filter",          bool_t, 0,  "enable/disable median filter",  False)
#gen.add("average_filter",         bool_t, 0,  "enable/disable average filter",  False)
#gen.add("align",                  bool_t, 0,  "adjust alignment",  False)

exit(gen.generate(PACKAGE, "c320_camera_node", "vsemi_c320_camera"))
